File: tests/pywin/win32com_/test/testWMI.py
from win32com_.client import GetObject
import win32com_.test.util

import unittest


class Simple(win32com_.test.util.TestCase):
    def testit(self):
        cses = GetObject("WinMgMts:").InstancesOf("Win32_Process")
        vals = []
        for cs in cses:
            val = cs.Properties_("Caption").Value
            vals.append(val)
        self.assertFalse(len(vals) < 5, "We only found %d processes!" % len(vals))


if __name__ == "__main__":
    unittest.main()

File: tests/pywin-py2many/win32com_/test/testWMI.jl
using win32com_.client: GetObject
import win32com_.test.util

abstract type AbstractSimple <: Abstractwin32com_.test.util.TestCase end
mutable struct Simple <: AbstractSimple
end
function testit(self::Simple)
    cses = InstancesOf(GetObject("WinMgMts:"), "Win32_Process")
    vals = []
    for cs in cses
        val = Properties_(cs, "Caption").Value
        push!(vals, val)
    end
    assertFalse(self, length(vals) < 5, "We only found %d processes!" % length(vals))
end

if abspath(PROGRAM_FILE) == @__FILE__
end
